### PR TITLE
Fix unexpected time zone shift of received timestamps when running th…

### PR DESCRIPTION
**Pull request by cyulysses-corp**
Fix unexpected time zone shift of received timestamps when running the instance from non-UTC zones.
b/227306541:

If run from a machine not in UTC time zone, the received UTC timestamps are different from received from the database. For example:
```
java -cp ExtractionTool_deploy.jar:terajdbc4.jar com/google/cloud/bigquery/dwhassessment/extractiontool/ExtractionTool td-extract --db-address jdbc:teradata://localhost/DBS_PORT=1025,DATABASE=dbc --output tmp_testTimestamp --db-user dbc --db-password dbc --rows-per-chunk 100 --sql-scripts querylogs --qrylog-timerange-start 2022-03-24T12:39:42.500000 --qrylog-timerange-end 2022-03-26T21:52:01.680000 --no-need-querytext --no-need-jdbc-schemas
```
Using --dry-run we can confirm that the sql script queries with the time range WHERE "DBC"."QryLogV"."StartTime" BETWEEN TIMESTAMP '2022-03-24 12:39:42.500000+00:00' AND TIMESTAMP '2022-03-26 21:52:01.680000+00:00' Directly running this query in the BTEQ console of our test Teradata database returns results with the first StartTime as 2022-03-24 12:39:42.500000+00:00 and the last StartTime as 2022-03-26 21:52:01.680000+00:00, as expected.

However, looking at the resulted .avro records, we see that the first entry's timestamp in milliseconds is 1648121982500, while the last one is 1648327921680, both 1 hour prior to the correct results (all in regard to UTC). The parsed UTC timestamps in the filename agree with the saved timestamps: querylogs-20220324T113942S500000-20220326T205201S680000_0.avro (both 1 hour off the supposedly received UTC values).

The culprit is likely with JDBC's ResultSet.getTimestamp() method and/or its Teradata implementation: by getting a timestamp from TIMESTAMP WITH TIMEZONE value, we should receive both the adjusted timestamp and the timezone it is adjusted towards; I have checked that the received timezone info is correct ("GMT+00:00",offset=0,dstSavings=0,useDaylight=false,transitions=0,lastRule=null), so it is mysterious why the timestamp . It seems like somewhere in the JDBC API the program thinks the timestamp is in GMT+1 (I'm running it from Warsaw locale) and forces dialling them back one hour.

### CODE REVIEW

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java [hunk 1]
```diff
@@ -33,6 +34,7 @@
 import java.sql.Timestamp;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java:37]: the import seems to be unused

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java [hunk 2]
```diff
@@ -168,6 +173,33 @@ private static void convertColumnTypeToAvroType(
...
+  public static Timestamp getUnadjustedTimestamp(ResultSet row, String columnName)
+      throws SQLException {
+    Calendar cal = Calendar.getInstance();
+    Timestamp timestamp = row.getTimestamp(columnName, cal);
+    return unadjustTimestamp(timestamp, cal);
+  }
+
+  public static Timestamp getUnadjustedTimestamp(ResultSet row, int columnIndex)
+      throws SQLException {
+    Calendar cal = Calendar.getInstance();
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java:185]: It seems this line can be moved under the `unadjustTimestamp`

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java [hunk 3]
```diff
@@ -168,6 +173,33 @@ private static void convertColumnTypeToAvroType(
     }
   }
 
+  public static Timestamp getUnadjustedTimestamp(ResultSet row, String columnName)
```
Comment (frsv) [inline src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java:176]: Please, add javadoc for these 2 methods
Also, this definitely requires a test. Could you add it?

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java (hunk 2 above)
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java:185]: Actually no, row.getTimestamp(columnIndex, cal) and row.getTimestamp(columnName, cal) actually modify the cal by setting its time zone (surprise, surprise, another anti-pattern), and this time zone information is needed in the unadjustTimestamp()

Comment (frsv) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java:185]: My bad, I didn't notice for some reason that it is also passed to `row.getTimestamp`

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java (hunk 1 above)
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/ScriptManagerImpl.java]: Done.

File: src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java (hunk 3 above)
Comment (cyulysses-corp) [reply src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/db/AvroHelper.java:186]: Added javadocs.

Also added unit tests. Ran manual tests too:
```
java -cp ExtractionTool_deploy.jar:terajdbc4.jar com/google/cloud/bigquery/dwhassessment/extractiontool/ExtractionTool td-extract --db-address jdbc:teradata://localhost/DBS_PORT=1025,DATABASE=dbc --output tmp_testTimestamp --db-user dbc --db-password dbc --rows-per-chunk 100 --sql-scripts querylogs --qrylog-timerange-start 2022-03-24T12:39:42.500000 --qrylog-timerange-end 2022-03-26T21:52:01.680000 --no-need-querytext --no-need-jdbc-schemas
```
verifying the timestamps are correct.

**Review by frsv — APPROVED** [review]

File: integ-tests/teradata/src/main/java/com/google/tdjdbc/JdbcUtil.java [hunk 4]
```diff
@@ -91,7 +92,11 @@ public static double getDoubleNotNull(ResultSet rs, String column) throws SQLExc
    * @return long or 0L if null.
    */
   public static long getTimestampNotNull(ResultSet rs, String column) throws SQLException {
-    Timestamp timestamp = rs.getTimestamp(column);
-    return rs.wasNull() ? 0L : timestamp.getTime();
+    Calendar cal = Calendar.getInstance();
+    Timestamp timestamp = rs.getTimestamp(column, cal);
+    if (rs.wasNull()) return 0L;
```
Comment (haselbach) [inline integ-tests/teradata/src/main/java/com/google/tdjdbc/JdbcUtil.java:97]: Can you run `git fix` please to ensure the formatting is according to the style guide? This line should violate https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used

**Review by haselbach — APPROVED** [review]

File: integ-tests/teradata/src/main/java/com/google/tdjdbc/JdbcUtil.java (hunk 4 above)
Comment (cyulysses-corp) [reply integ-tests/teradata/src/main/java/com/google/tdjdbc/JdbcUtil.java]: I did run google-java formatter repeatedly and strangely it doesn't correct this according to the javaguide. Should we report this bug?

**Pull request merged**